components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -418,18 +418,20 @@ private void handleIdPInitSSO(HttpServletRequest req, HttpServletResponse resp,
                             SAMLSSOServiceProviderDO serviceProviderDO = getSPConfig(SAMLSSOUtil
                                             .getTenantDomainFromThreadLocal(),
                                     SAMLSSOUtil.splitAppendedTenantDomain(issuer));
-                            //for IDP init SLO, priority should be given to SLO response URL over default ACS.
-                            acsUrl = serviceProviderDO.getSloResponseURL();
-                            if (StringUtils.isBlank(acsUrl)) {
-                                acsUrl = serviceProviderDO.getDefaultAssertionConsumerUrl();
-                            }
+                            if (serviceProviderDO != null) {
```

Comment:
What happens if ServiceProviderDO is null?
Do we throw an exception and stop the flow?

Reply:
here acsUrl is required for an error scenario path. serviceProviderDO is obtained to set the acsUrl as a query param to saml notification.do page if it is null. This shouldn't break the flow.